evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -377,15 +377,17 @@ where
                 let ctx = MockCallDataContext {
                     calldata: call.input.clone(),
                     value: Some(call.transfer.value),
+                    reverts: false
```

Comment:
can we please document was this the following logic does, because this gets a bit confusing.

we effectively checking if there's a non reverting call, if so we return.
if not, then we find the by checking the calldata